python/HeaderService/hslib_salobj.py

```diff
@@ -155,12 +155,75 @@ def get_ip(self):
         self.ip_address = socket.gethostbyname(socket.gethostname())
         self.log.info("Will use IP: {} for web service".format(self.ip_address))
 
+    def get_s3instance(self):
+        """
+        Figure the location where are running to define the s3instance
+        in case it wasn't defined.
+        """
+        # Check if defined in self.config or the environment
+        if self.config.s3instance:
+            s3instance = self.config.s3instance
+            self.log.info(f"Will use s3instance in config: {s3instance}")
+            return s3instance
+        elif 'S3INSTANCE' in os.environ:
+            s3instance = os.environ['S3INSTANCE']
+            self.log.info(f"Will use s3instance from environment: {s3instance}")
+            return s3instance
+
+        # Try to auto-figure out from location
+        self.log.warning("The s3instance was not defined in config")
```

Comment:
Note that the auto-figure option is a last resource option that uses `socket.getfqdn()` and it is only triggered if **neither** `self.config.s3instance` (from the yaml file) or the env variable `S3INSTANCE` are defined. I'm happy to remove the auto-configure part and just let it fail if neither `S3INSTANCE` or `self.config.s3instance` are not defined.

Reply:
Sorry. I guess I was confused on that part. The one issue I have with the config file is that begins to imply that the HS is a configurable CSC which currently it is not. However since the ENVVAR is the backup, I guess I'm less concerned about the configuration file.

Reply:
Actually, by default the `s3instance` in not defined in the configuration file, so it need to be either set by the command-line argument `--s3instance` or the environment.